exp/tools/serpapi/internal/serpapiClient.go

```diff
@@ -0,0 +1,108 @@
+package internal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+var ErrMissingToken = errors.New("missing the OpenAI API key, set it in the OPENAI_API_KEY environment variable")
+
+type SerpapiClient struct {
+	params url.Values
+}
+
+func New() (*SerpapiClient, error) {
+	apiKey := os.Getenv("SERPAPI_API_KEY")
+	if apiKey == "" {
+		return nil, ErrMissingToken
+	}
+
+	params := url.Values{}
+	params.Add("google_domain", "google.com")
+	params.Add("gl", "us")
+	params.Add("hl", "en")
+	params.Add("api_key", apiKey)
+	return &SerpapiClient{
+			params: params,
+		},
+		nil
```

Comment:
This part of code didn't indent correctly, can you run `go fmt ./...` and commit again?